### PR TITLE
Update api

### PR DESCRIPTION
**Pull request by qwrmolly**
requests for meerkatv2 to detect logs and console.logs return("") instead of console.logs("")

approval:
**### _approved_** by WindowsMeosu

### CODE REVIEW

File: Request/API/api [hunk 1]
```diff
@@ -13,20 +14,23 @@ return function
...
 retrive log.success
 function(find log.success; log.config(meerkatv1))
 if failed;(return log.success(create log.connection_failed)
-return log.config; log.connection_failed;console.logs(run.log.config(""); console.logs("connection failed, retrying") 
-console.logs("attempting connection to cho.ppy.sh")
+return log.config; log.connection_failed;console.logs(run.log.config("");console.logs return("connection failed, retrying")
+ close request)
+request(
+ console.logs return("attempting connection to cho.ppy.sh")
  retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
```
Comment (WindowsMeosu) [inline Request/API/api:29]:
```suggestion
 retrive_path(/request/api/irc/irc); request_path(/request/api/irc/irc)
```

File: Request/API/api [hunk 2]
```diff
@@ -13,20 +14,23 @@ return function
 logsv1; logsv2; logsv3:(console.logs)
 return function
 logsv1; logsv2; logsv3; logsv4:(console.logs.delete)
-console.logs("connected to meerkatv2")
-console.logs("attempting connection to cho.ppy.sh")
- retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
+console.logs return("connected to meerkatv2")
+console.logs return("attempting connection to cho.ppy.sh")
+retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
```
Comment (WindowsMeosu) [inline Request/API/api:19]:
```suggestion
retrive_path(/request/api/irc/irc); request_path(/request/api/irc/irc)
```

**Review by WindowsMeosu — CHANGES_REQUESTED** [review]
requesting following changes

File: Request/API/api [hunk 3]
```diff
@@ -1,6 +1,7 @@
 config :osu_api, api_key: retrive:API_key
 
 osu_api(retrive:total_score);(retrive:ranked_score);(retrive:get_user(user, user_id))
+request(
 
  function(retrive(logsv1)(logsv2)(logsv3)(logsv4)(logsv5)(logsv6)(logsv7)(logsv8)(logsv9)(logsv10)(logsv11)(logsv12)(logsv13))
```
Comment (WindowsMeosu) [inline Request/API/api:6]:
```suggestion
function retrieve .logs
```
requesting that you change this also, considering that I didn't knew what I was thinking when multiple of the same logs is sent to console.logs.

File: Request/API/api [hunk 4]
```diff
@@ -1,6 +1,7 @@
 config :osu_api, api_key: retrive:API_key
 
 osu_api(retrive:total_score);(retrive:ranked_score);(retrive:get_user(user, user_id))
```
Comment (WindowsMeosu) [inline Request/API/api:3]:
```suggestion
osu_api(retrieve :total_score);(retrieve :ranked_score);(retrieve :get_user(user, user_id))
```

File: Request/API/api (hunk 3 above)
Comment (WindowsMeosu) [inline Request/API/api:6]: function retrieve(.logs; listusers(meerkatv1)
retrieves user list from meerkatv1
and adds it to .listuserslogs

File: Request/API/api (hunk 1 above)
Comment (WindowsMeosu) [inline Request/API/api:29]:
```suggestion
retrieve_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
```

File: Request/API/api [hunk 5]
```diff
@@ -13,20 +14,23 @@ return function
 logsv1; logsv2; logsv3:(console.logs)
```
Comment (WindowsMeosu) [inline Request/API/api:14]:
```suggestion
.logs_messages; .logs_users; .listuserslogs:(console.logs)
```

File: Request/API/api [hunk 6]
```diff
@@ -13,20 +14,23 @@ return function
...
 return function
 logsv1; logsv2; logsv3; logsv4:(console.logs.delete)
```
Comment (WindowsMeosu) [inline Request/API/api:16]:
```suggestion
.logs_messages; .logs_users; .listuserslogs:(console.logs.delete)
```

File: Request/API/api [hunk 7]
```diff
@@ -13,20 +14,23 @@ return function
...
 if failed;(return log.success(create log.connection_failed)
-return log.config; log.connection_failed;console.logs(run.log.config(""); console.logs("connection failed, retrying") 
-console.logs("attempting connection to cho.ppy.sh")
+return log.config; log.connection_failed;console.logs(run.log.config("");console.logs return("connection failed, retrying")
+ close request)
+request(
+ console.logs return("attempting connection to cho.ppy.sh")
  retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
 retrive log.config
 retrive log.success
```
Comment (WindowsMeosu) [inline Request/API/api:31]:
```suggestion
retrieve log.success
```
fixed all spelling mistakes (retrive - retrieve) 'hopefully'

File: Request/API/api [hunk 8]
```diff
@@ -13,20 +14,23 @@ return function
...
 function(find log.success; log.config(meerkatv1))
 if failed;(return log.success(create log.connection_failed)
-return log.config; log.connection_failed;console.logs(run.log.config(""); console.logs("connection failed, retrying") 
-console.logs("attempting connection to cho.ppy.sh")
+return log.config; log.connection_failed;console.logs(run.log.config("");console.logs return("connection failed, retrying")
+ close request)
+request(
+ console.logs return("attempting connection to cho.ppy.sh")
  retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
 retrive log.config
```
Comment (WindowsMeosu) [inline Request/API/api:30]:
```suggestion
retrieve log.config
```

File: Request/API/api [hunk 9]
```diff
@@ -13,20 +14,23 @@ return function
...
 logsv1; logsv2; logsv3; logsv4:(console.logs.delete)
-console.logs("connected to meerkatv2")
-console.logs("attempting connection to cho.ppy.sh")
- retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
+console.logs return("connected to meerkatv2")
+console.logs return("attempting connection to cho.ppy.sh")
+retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
 retrive log.config
 function(run.log.config)
 retrive log.success
```
Comment (WindowsMeosu) [inline Request/API/api:22]:
```suggestion
retrieve log.success
```

File: Request/API/api [hunk 10]
```diff
@@ -13,20 +14,23 @@ return function
 logsv1; logsv2; logsv3:(console.logs)
 return function
 logsv1; logsv2; logsv3; logsv4:(console.logs.delete)
-console.logs("connected to meerkatv2")
-console.logs("attempting connection to cho.ppy.sh")
- retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
+console.logs return("connected to meerkatv2")
+console.logs return("attempting connection to cho.ppy.sh")
+retrive_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
 retrive log.config
```
Comment (WindowsMeosu) [inline Request/API/api:20]:
```suggestion
retrieve log.config
```

File: Request/API/api (hunk 1 above)
Comment (WindowsMeosu) [inline Request/API/api:29]:
```suggestion
 retrieve_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
```

File: Request/API/api (hunk 7 above)
Comment (WindowsMeosu) [inline Request/API/api:31]:
```suggestion
retrieve log.success
```

**Review by WindowsMeosu — CHANGES_REQUESTED** [review]

**Review by WindowsMeosu — APPROVED** [review]
after you add all those changes, we'll merge.

File: Request/API/api (hunk 3 above)
Comment (qwrmolly) [reply Request/API/api]: dc

File: Request/API/api (hunk 2 above)
Comment (qwrmolly) [reply Request/API/api:19]:
```suggestion
retrieve_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
```
confused here.. run_path will return /request/api/irc/irc faster and errors sent to meerkatv1/console.logs however request_logs will return it slower and won't errors to meerkatv1/console/logs nor meerkatv2/console.logs.

File: Request/API/api (hunk 1 above)
Comment (qwrmolly) [reply Request/API/api:29]:
```suggestion
retrieve_path(/request/api/irc/irc); run_path(/request/api/irc/irc)
```
read https://github.com/realMiloTilo/osu-meerkat/pull/33/files/ac88a4c231916036b161f1b2f470eaadc4fac96a#r973623365

Comment (qwrmolly) [reply Request/API/api]: duplicate
not commiting

File: Request/API/api (hunk 7 above)
Comment (qwrmolly) [reply Request/API/api]: dupe

File: Request/API/api (hunk 3 above)
Comment (qwrmolly) [inline Request/API/api]:
```suggestion
 function retrieve(.logs; listusers(meerkatv1)
 find (.logs; listuserslogs(meerkatv1)
```

Comment (WindowsMeosu) [reply Request/API/api]: yeah that can work
will improve once we change it to a actual script that won't have any issues

Comment (WindowsMeosu) [reply Request/API/api]: since you already added suggested changes to that, I'll use that if current changes break

File: Request/API/api (hunk 2 above)
Comment (WindowsMeosu) [reply Request/API/api:19]: meh

fair

File: Request/API/api [hunk 11]
```diff
@@ -1,32 +1,36 @@
 config :osu_api, api_key: retrive:API_key
```
Comment (WindowsMeosu) [inline Request/API/api:1]:
```suggestion
config :osu_api, api_key: retrieve:API_key
```
I'll add myself

Comment (WindowsMeosu) [reply Request/API/api:1]: spelling mistakes fixed

**Pull request merged**